src/lti/LTI_OIDC_Login.php

```diff
@@ -29,16 +29,22 @@ function __construct(Database $database, Cache $cache = null, Cookie $cookie = n
 
     /**
      * Static function to allow for method chaining without having to assign to a variable first.
+     *
+     * @param Database $database Instance of the database interface used for looking up registrations and deployments.
+     * @param Cache    $cache    Instance of the Cache interface used to loading and storing launches. If non is provided launch data will be store in $_SESSION.
+     * @param Cookie   $cookie   Instance of the Cookie interface used to set and read cookies. Will default to using $_COOKIE and setcookie.
+     * 
+     * @return self
      */
-    public static function new(Database $database, Cache $cache = null, Cookie $cookie = null) {
+    public static function newInstance(Database $database, Cache $cache = null, Cookie $cookie = null) {
         return new LTI_OIDC_Login($database, $cache, $cookie);
     }
 
     /**
      * Calculate the redirect location to return to based on an OIDC third party initiated login request.
      *
-     * @param string        $launch_url URL to redirect back to after the OIDC login. This URL must match exactly a URL white listed in the platform.
-     * @param array|string  $request    An array of request parameters. If not set will default to $_REQUEST.
+     * @param string       $launch_url URL to redirect back to after the OIDC login. This URL must match exactly a URL white listed in the platform.
+     * @param array|string $request    An array of request parameters. If not set will default to $_REQUEST.

```

Comment:
Not you, obviously, but I don't see how this could be a `string` - it's treated like an array in the body.